airflow/providers/microsoft/azure/hooks/wasb.py

```diff
@@ -385,7 +385,7 @@ def upload(
         return blob_client.upload_blob(data, blob_type, length=length, **kwargs)
 
     def download(
-        self, container_name, blob_name, offset: int, length: int, **kwargs
+        self, container_name, blob_name, offset: int | None = None, length: int | None = None, **kwargs
```

Comment:
This is what we had before https://github.com/apache/airflow/pull/25426
we needed to make the change to update to newer sdk version
see https://github.com/apache/airflow/pull/25426#issuecomment-1233152203

Reply:
I am afraid #25426 was a mistake. 

They simply added type-hints but left None as default. They simply use "implict optional" and we don't use it.

Implicit optional is a setting in https://github.com/apache/airflow/blob/1eaedc8ae85dbbfaaa402f9936304be51de98f88/setup.cfg#L191 to add azure-storage to list of other azure libraries.

Reply:
ah! I didn't know about this setting

Reply:
Still @malthe you will need to add azure-storage to setup.cfg

Reply:
This may be the solution for other azure packages https://github.com/apache/airflow/pull/27199

Reply:
What's the deal with those exclusions:
```
[mypy-azure.batch.*]
no_implicit_optional = False
[mypy-azure.batch.models.*]
no_implicit_optional = False
```
Does it pattern-match on the module where the problem is? I am not totally sure what to put here for azure-storage, @potiuk.
```cfg
[mypy-azure.storage.*]
no_implicit_optional = False
```
^^^ something like that?

Reply:
I think it's best to consult the docs to be sure: https://mypy.readthedocs.io/en/stable/config_file.html#examples . 

I can't remember all nuances, but I beleve this is the package whic azure storage provides.

Reply:
The docs say:

> Some of the config options may be set either globally (in the [mypy] section) or on a per-module basis (in sections like [mypy-foo.bar]).

Reply:
But ultimately what I usually do is to add what I think works and check (run mypy static checks in this case :) )

Reply:
> This may be the solution for other azure packages #27199

Likely yes.

Reply:
From the docs it seems like a mistake to have the same rule for both "azure.batch" and "azure.batch.models". I am not totally sure what ".*" adds to the picture. The docs aren't really clear (to me at least).

Reply:
Try to remove and see if it works with `--all-files` :)

If it's not clear, just testing hypothesis is a good idea.

Reply:
Unfortunately, none of us can be made responsible for clarity of docs that we have little control over, but we have control on verifying our assumptions in case they are unclear.

Reply:
(and yes we can complain after we do).

Reply:
Fixed in d77e6815b9c586b6d8aa4cc72120727854a77d6e. I ended up trading one line for another, perhaps not the cleanest commit ever.